pkg/events/definition_dependencies.go

```diff
@@ -13,6 +13,8 @@ type Dependencies struct {
 	probes       []Probe
 	tailCalls    []TailCall
 	capabilities Capabilities
+	// fallbacks contains alternative dependency sets to try if the primary dependencies fail
+	fallbacks []Dependencies
```

Comment:
Having fallbacks inside the fallback dependencies doesn't make sense.
Instead, we can wrap Dependencies in another struct that will contain the "original" dependencies and a list of fallback Dependencies